src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContextFactory.cs

```diff
@@ -14,15 +15,21 @@ namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
     /// </summary>
     internal interface IWorkspaceProjectContextFactory
     {
+        /// <inheritdoc cref="CreateProjectContextAsync"/>
+        [Obsolete("Use CreateProjectContextAsync instead")]
+        IWorkspaceProjectContext CreateProjectContext(string languageName, string projectUniqueName, string projectFilePath, Guid projectGuid, object hierarchy, string binOutputPath);
+
         /// <summary>
-        /// Creates and initializes a new Workspace project and returns a <see cref="IWorkspaceProjectContext"/> to lazily initialize the properties and items for the project.
+        /// Creates and initializes a new Workspace project and returns a <see cref="IWorkspaceProjectContext"/> to
+        /// lazily initialize the properties and items for the project.  Cancellation is supported and will not leave
+        /// the workspace in a partially completed state.
```

Comment:
```suggestion
        /// lazily initialize the properties and items for the project.  Cancellation is supported and will not leave
        /// the workspace in a partially completed state.
```

I think I'd rephrase that slightly to say something more like:

> This method guarantees that either the project is added (and the returned task completes) or cancellation is observed and no project is added.

Reply:
done. thanks.